lib/langchain/tool/database.rb

```diff
@@ -0,0 +1,43 @@
+module Langchain::Tool
+  class Database < Base
+    description <<~DESC
+      Useful for getting the result of a database query.
+
+      The input to this tool should be valid SQL.
+    DESC
+
+    # Establish a database connection
+    # example: 'postgres://postgres:postgres@localhost:5432/db_name'
+    # Empty string creates memory database using sqlite3
+    def initialize(db_connection_string = "")
+      depends_on "sequel"
+      depends_on "sqlite3"
+      require "sequel"
+      require "sequel/extensions/schema_dumper"
+
+      @db = if db_connection_string.empty?
+        Sequel.sqlite
+      else
+        Sequel.connect(db_connection_string)
+      end
+      @db.extension :schema_dumper
+    end
+
+    def schema
+      Langchain.logger.info("Database: Getting schema")
+      @db.dump_schema_migration(same_db: true, foreign_keys: false, indexes: false)
```

Comment:
I think we need to tell the LLM about foreign keys because otherwise it wouldn't know how to join tables together.

Reply:
Sounds good.

Reply:
How difficult would it be to change this to match the schema format that more resembles: https://github.com/andreibondarev/langchainrb/pull/91#issuecomment-1574592840

```ruby
tables = @db.tables
@db["SELECT table_name, column_name, data_type FROM information_schema.columns where table_name in (#{tables})"] # Needs to be reformatted to match ('users', 'user_preferences') etc.
```

Reply:
I think I would have to write that query for every database supported by Sequel, or should I just try it for Postgres for now?

Reply:
@mattlindsey If it's okay -- can we just do it for Postgres for now? Other DB vendors in separate issues.